stepconf/stepconf.go

```diff
@@ -171,6 +187,260 @@ func setField(field reflect.Value, value, constraint string) error {
 	return nil
 }
 
+//ValidateRangeFields validates if the given range is proper.
+func ValidateRangeFields(valueStr, constraint string) error {
+	constraintMin, constraintMax, constraintMinBr, constraintMaxBr, err := GetRangeValues(constraint)
+	if err != nil {
+		return err
+	}
+	min, err := parseValueStr(constraintMin)
+	if err != nil {
+		return fmt.Errorf("failed to parse min value: %s", err)
+	}
+	max, err := parseValueStr(constraintMax)
+	if err != nil {
+		return fmt.Errorf("failed to parse max value: %s", err)
+	}
+	value, err := parseValueStr(valueStr)
+	if err != nil {
+		return fmt.Errorf("failed to parse value: %s", err)
+	}
+	isMinInclusiveBool, err := isMinInclusive(constraintMinBr)
+	if err != nil {
+		return err
+	}
+	isMaxInclusiveBool, err := isMaxInclusive(constraintMaxBr)
+	if err != nil {
+		return err
+	}
+
+	if err := validateRangeFieldValues(min, max, isMinInclusiveBool, isMaxInclusiveBool, value); err != nil {
+		return err
+	}
+	if err := validateRangeFieldTypes(min, max, value); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isMinInclusive(bracket string) (bool, error) {
+	if bracket == "[" {
```

Comment:
please use a switch with a default element so you no need to have a return at the end

Reply:
Updated